suite-common/wallet-core/src/fiat-rates/fiatRatesSelectors.ts

```diff
@@ -0,0 +1,211 @@
+        ) as {
+            account: Account;
+            txs: WalletAccountTransaction[];
+        }[];
+    },
+    // There could be only one active local currency at the time
+    { size: 1 },
+);
+
+/**
+ * @deprecated Use selectFiatRatesByFiatRateKey or any other selector
+ */
+export const selectCoinsLegacy = memoize(
+    (state: FiatRatesRootState): FiatRatesStateLegacy['coins'] => {
+        const coins: FiatRatesStateLegacy['coins'] = [];
+
+        Object.values(state.wallet.fiat.current).forEach(rate => {
+            const coin = coins.find(
+                c => c.symbol === rate.ticker.symbol && c.tokenAddress === rate.ticker.tokenAddress,
+            );
+            if (coin && coin.current) {
+                coin.current.rates[rate.locale] = rate.rate;
+            } else {
+                coins.push({
+                    ...rate.ticker,
+                    current: {
+                        rates: {
+                            [rate.locale]: rate.rate,
+                        },
+                        ...rate.ticker,
+                        ts: rate.lastSuccessfulFetchTimestamp,
+                    },
+                });
+            }
+        });
+
+        Object.values(state.wallet.fiat.lastWeek).forEach(rate => {
+            const coin = coins.find(
+                c => c.symbol === rate.ticker.symbol && c.tokenAddress === rate.ticker.tokenAddress,
+            );
+            if (coin) {
+                const ticker = coin.lastWeek?.tickers.find(
+                    t => t.ts === rate.lastSuccessfulFetchTimestamp,
+                );
+                if (ticker) {
+                    ticker.rates[rate.locale] = rate.rate;
+                } else {
+                    coin.lastWeek?.tickers.push({
+                        rates: {
+                            [rate.locale]: rate.rate,
+                        },
+                        ts: rate.lastSuccessfulFetchTimestamp,
+                    });
+                }
+            } else {
+                coins.push({
+                    ...rate.ticker,
+                    lastWeek: {
+                        tickers: [
+                            {
+                                rates: {
+                                    [rate.locale]: rate.rate,
+                                },
+                                ts: rate.lastSuccessfulFetchTimestamp - ONE_WEEK_IN_MS,
+                            },
+                        ],
+                        ts: rate.lastSuccessfulFetchTimestamp,
+                        ...rate.ticker,
+                    },
+                });
+            }
+        });
+
+        return coins;
+    },
+);
+
+export const selectIsAccountWithRatesByKey = memoizeWithArgs(
+    (state: AccountsRootState & FiatRatesRootState, accountKey: string) => {
+        const account = selectAccountByKey(state, accountKey);
+
+        if (!account) {
+            return false;
+        }
+
+        // TODO: refactor
```

Comment:
It would be great to have some more info about what is wrong and what shall be refactored

Reply:
The idea is to use `selectFiatRatesByFiatRateKey` as mobile does

Reply:
Ok, but I think lets add this note into the comment there I guess.

Reply:
Or maybe better - I don't really know it just think it, this is taken from native (done by @Nodonisko)

Reply:
I like to discuss this more, but for now it is not blocking the merge. Feel free to resolve if you really don't want to fix it now.